src/integrators/whittedIntegrator.cpp

```diff
@@ -88,11 +88,11 @@ Spectrum* WhittedIntegrator::integrate(const Ray& ray) {
       delete reflectedRay.origin;
       delete spec;
     }
-    delete sample;
```

Comment:
in case we revert the change, we should move this `delete sample` into the sample->isValid if-block.